### PR TITLE
Add more logging to methods responsible for simulator startup and split logic between Xcode9 and earlier versions

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Made the code more readable and easier to debug from logs

### CODE REVIEW

File: lib/simulator-xcode-6.js [hunk 1]
```diff
@@ -272,59 +273,60 @@ class SimulatorXcode6 extends EventEmitter {
...
+  async run (opts = {}) {
+    opts = Object.assign({
       allowTouchEnroll: false,
       startupTimeout: this.startupTimeout,
     }, opts);
+    const {state} = await this.stat();
+    const isServerRunning = state === 'Booted';
+    const isUIClientRunning = await this.isUIClientRunning();
+    if (isServerRunning && isUIClientRunning) {
+      log.info('Both Simulator with UDID ${this.udid} and the UI client are currently running. There\'s no need for restart.');
```
Comment (imurchie) [inline lib/simulator-xcode-6.js]: This needs to be in backtick quotes to get the template insertion. Then the escape of the apostrophe is also unneeded.

File: lib/simulator-xcode-9.js [hunk 2]
```diff
@@ -16,39 +17,73 @@ class SimulatorXcode9 extends SimulatorXcode8 {
...
+      }
+      if (wasUIClientKilled) {
+        // Stopping the UI client also kills all running servers. Sad but true
+        log.info(`Detected the UI client was running and killed it. Verifying the Simulator is in Shutdown state...`);
+        await waitForShutdown();
+      }
+      await bootSimulator();
+    } else {
+      if (isServerRunning && isUIClientRunning) {
+        log.info('Both Simulator with UDID ${this.udid} and the UI client are currently running.');
```
Comment (imurchie) [inline lib/simulator-xcode-9.js]: Template literal.

File: lib/simulator-xcode-6.js (hunk 1 above)
Comment (mykola-mokhnach) [inline lib/simulator-xcode-6.js]: thanks, fixed

File: lib/simulator-xcode-9.js (hunk 2 above)
Comment (mykola-mokhnach) [inline lib/simulator-xcode-9.js]: thanks, fixed

**Pull request merged**